extensions/twitch.js

```diff
@@ -0,0 +1,297 @@
+          },
+          {
+            opcode: "closeimage",
+            text: "close image",
+            blockType: Scratch.BlockType.COMMAND,
+            blockIconURI: "https://tekinical.github.io/twitchURI.png",
+          },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Utilities",
+          },
+          {
+            opcode: "true",
+            text: "true",
+            blockType: Scratch.BlockType.BOOLEAN,
+            blockIconURI: "https://tekinical.github.io/twitchURI.png",
+          },
+          {
+            opcode: "false",
+            text: "false",
+            blockType: Scratch.BlockType.BOOLEAN,
+            blockIconURI: "https://tekinical.github.io/twitchURI.png",
+          },
+        ],
+      };
+    }
+
+    setupchat(args) {
+      var Link =
+        "https://nightdev.com/hosted/obschat/?theme=bttv_dark&channel=" +
+        args.username +
+        "&fade=" +
+        args.fade +
+        "&bot_activity=" +
+        args.bots +
+        "&prevent_clipping=" +
+        args.prevent;
+      chat = document.createElement("iframe");
+      chat.style.width = args.width;
+      chat.style.height = args.height;
+      chat.style.border = "none";
+      chat.style.position = args.position;
+      chat.setAttribute("allowtransparency", "true");
+      chat.setAttribute("src", Link);
+      Scratch.renderer.addOverlay(chat, "manual");
+    }
+
+    closechat() {
+      if (chat == null) {
+        console.log("Can't close chat if chat wasn't opened.");
+      } else {
+        Scratch.renderer.removeOverlay(chat);
+        chat = null;
+      }
+    }
+
+    text(args) {
+      if (text == null) {
+        text = document.createElement("p");
+        text.style.font = args.font;
+        text.innerText = args.text;
+        text.width = "20%";
+        text.height = "10%";
+        text.style.border = "none";
+        text.style.position = "absolute";
+        text.style.bottom = 0;
+        if (args.bold) {
+          text.style.fontWeight = "bold";
+        }
+        if (args.italic) {
+          text.style.fontStyle = "italic";
+        }
+        text.style.fontSize = args.size;
+        text.style.fontFamily = args.font;
+        Scratch.renderer.addOverlay(text, "manual");
+      } else {
+        Scratch.renderer.removeOverlay(text);
+        text = document.createElement("p");
+        text.style.font = args.font;
+        text.innerText = args.text;
+        text.width = "20%";
+        text.height = "10%";
+        text.style.border = "none";
+        text.style.position = "absolute";
+        text.style.bottom = 0;
+        if (args.bold) {
+          text.style.fontWeight = "bold";
+        }
+        if (args.italic) {
+          text.style.fontStyle = "italic";
+        }
+        text.style.fontSize = args.size;
+        text.style.fontFamily = args.font;
+        Scratch.renderer.addOverlay(text, "manual");
+      }
+    }
+
+    image(args) {
```

Comment:
There is duplicate code here, I'm sure you can find a way to shorten this.

Reply:
Done!